core/rfb.js

```diff
@@ -1307,33 +1333,36 @@ export default class RFB extends EventTargetMixin {
                     case 'longpress':
                         this._fakeMouseMove(ev, pos.x, pos.y);
                         break;
-                    case 'twodrag':
+                    case 'twodrag': {
+                        let bmask = RFB._convertButtonMask(ev.detail.buttons);
                         // Always scroll in the same position.
                         // We don't know if the mouse was moved so we need to move it
                         // every update.
                         this._fakeMouseMove(ev, pos.x, pos.y);
                         while ((ev.detail.magnitudeY - this._gestureLastMagnitudeY) > GESTURE_SCRLSENS) {
-                            this._handleMouseButton(pos.x, pos.y, true, 0x8);
-                            this._handleMouseButton(pos.x, pos.y, false, 0x8);
+                            this._handleMouseButton(pos.x, pos.y, true, bmask | 0x8);
```

Comment:
You should probably rather keep using the known mouse state. I.e. `this._mouseButtonMask`.